service/src/main/java/com/commercetools/pspadapter/payone/notification/common/CaptureNotificationProcessor.java

```diff
@@ -28,37 +30,64 @@ public CaptureNotificationProcessor(TenantFactory tenantFactory, TenantConfig te
         super(tenantFactory, tenantConfig, transactionStateResolver);
     }
 
-        @Override
+    @Override
     protected boolean canProcess(final Notification notification) {
         return NotificationAction.CAPTURE.equals(notification.getTxaction());
     }
 
     @Override
     protected ImmutableList<UpdateAction<Payment>> createPaymentUpdates(final Payment payment,
                                                                         final Notification notification) {
-        final ImmutableList.Builder<UpdateAction<Payment>> listBuilder = ImmutableList.builder();
-        listBuilder.addAll(super.createPaymentUpdates(payment, notification));
+        final ImmutableList.Builder<UpdateAction<Payment>> actionsBuilder = ImmutableList.builder();
+        actionsBuilder.addAll(super.createPaymentUpdates(payment, notification));
 
         final List<Transaction> transactions = payment.getTransactions();
         final String sequenceNumber = toSequenceNumber(notification.getSequencenumber());
 
         return findMatchingTransaction(transactions, TransactionType.CHARGE, sequenceNumber)
-                .map(transaction -> listBuilder.build())
-                .orElseGet(() -> {
-                    final MonetaryAmount amount = MoneyImpl.of(notification.getPrice(), notification.getCurrency());
+                .map(transaction -> updateChargeTransactionState(transaction, notification, actionsBuilder))
```

Comment:
Why don't you pass `TransactionState.SUCCESS` instead? It's more explicit what you want in case of `capture`. And here you pass `notification`, but you just use it to get `TransactionState`, so pass `TransactionState` directly instead?

Reply:
> Why don't you pass TransactionState.SUCCESS instead? 

Unfortunately, Payone documentation is not clear enough in this point, so it might be that the notification arrives with state `Pending`, but not `Success`. Here even a special test case for this:
https://github.com/commercetools/commercetools-payone-integration/blob/3de9fe90be5fef4bd7feb01b2e2ea78f3518df69/service/src/test/java/com/commercetools/pspadapter/payone/notification/common/CaptureNotificationProcessorTest.java#L65-L76

Reply:
> And here you pass notification, but you just use it to get TransactionState, so pass TransactionState directly instead?

As for passing `notification` or `TransactionState`: i just wanted to simplify the function call, and avoid lambda line look too long like this:
```java
.map(transaction -> updateChargeTransactionState(transaction, notification.getTransactionStatus().getCtTransactionState(), actionsBuilder))
```

Also, this approach is cleaner because we delegate responsibility of transaction update to the function itself: we kind of don't care how `updateChargeTransactionState()` is actually implemented and what properties it requires from the _notification_. Our responsibility is to provide all necessary information. If in the future we have to extend `updateChargeTransactionState()` behavior adding new properties from the notification - we don't have to change the method signature, only the implementation.